pkg/skaffold/deploy/helm.go

```diff
@@ -401,18 +401,23 @@ func (h *HelmDeployer) deleteRelease(ctx context.Context, out io.Writer, r lates
 
 func (h *HelmDeployer) joinTagsToBuildResult(builds []build.Artifact, params map[string]string) (map[string]build.Artifact, error) {
 	imageToBuildResult := map[string]build.Artifact{}
-	for _, build := range builds {
-		imageToBuildResult[build.ImageName] = build
+	for _, b := range builds {
+		imageToBuildResult[b.ImageName] = b
 	}
 
 	paramToBuildResult := map[string]build.Artifact{}
 	for param, imageName := range params {
 		newImageName := util.SubstituteDefaultRepoIntoImage(h.defaultRepo, imageName)
-		build, ok := imageToBuildResult[newImageName]
+		b, ok := imageToBuildResult[newImageName]
 		if !ok {
-			return nil, fmt.Errorf("no build present for %s", imageName)
+			if len(builds) == 0 {
+				logrus.Warnf("no build artifacts present. Assuming skaffold deploy. Continuing with %s", imageName)
```

Comment:
I would make this a Debug statement instead.

Reply:
I would actually just remove this altogether.

```golang
if len(builds) > 0 {
  return nil, fmt.Errorf("no build present for %s", imageName)
}
b = build.Artifact{ImageName: imageName, Tag: imageName}
```

Reply:
@nkubala, users should be able to run `skaffold deploy` even if skaffold did not build any images in the deployment.

Reply:
@nkubala I wanted to print a warn or a debug message to just give users more insight in to what is happening.

Reply:
right, this would do that. I'm just proposing that you remove the log message since I don't really think it's necessary. the change I proposed is logically equivalent, just a bit easier to read IMO.

```
if we didn't find the build in the build map:
  * if build map is not empty -> error out because we couldn't find the specified artifact
  * continue, because we want to allow deploying without anything being built
```